src/test/scala/com/advancedtelematic/campaigner/actor/CampaignSchedulerSpec.scala

```diff
@@ -27,13 +28,13 @@ class CampaignSchedulerSpec extends ActorSpec[CampaignScheduler] with Campaigner
   }
 
   "campaign scheduler" should "trigger updates for each group" in {
-    val groups   = arbitrary[Set[GroupId]].sample.get
-    val campaign = createDbCampaignWithUpdate(groups = groups).futureValue
+    val groups   = arbitrary[NonEmptyList[GroupId]].sample.get
+    val campaign = createDbCampaignWithUpdate(maybeGroups = Some(groups)).futureValue
 
     val parent   = TestProbe()
 
     campaigns.scheduleGroups(campaign.id, groups).futureValue
-    groups.foreach { g => deviceRegistry.setGroup(g, arbitrary[Seq[DeviceId]].sample.get) }
+    groups.map{ g => deviceRegistry.setGroup(g, arbitrary[Seq[DeviceId]].sample.get) }
```

Comment:
why not keep foreach? foreach is usually used when the iteration is just for side effects

Reply:
There is no `foreach` for `NonEmptyList` 😕